client.go

```diff
@@ -163,14 +163,16 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		scanres = res.(*pb.ScanResponse)
 		results = append(results, scanres.Results...)
 
 		// TODO: The more_results field of the ScanResponse object was always
 		// true, so we should figure out if there's a better way to know when
 		// to move on to the next region than making an extra request and
 		// seeing if there were no results
-		for len(scanres.Results) != 0 {
+		for len(scanres.Results) != 0 &&
+				(numberOfRows == 0 || uint32(len(results)) < numberOfRows) {
```

Comment:
the default value `DefaultNumberOfRows = 128` is set here: https://github.com/tsuna/gohbase/blob/master/hrpc/scan.go#L64
I'm thinking the right way would be to check if the value is strictly positive when it passed as an option here: https://github.com/tsuna/gohbase/blob/master/hrpc/call.go#L211

Sorry that these values are all over the place in gohbase/hrpc.